feature/home/src/main/java/com/epmedu/animeal/home/presentation/ui/HomePermissions.kt

```diff
@@ -0,0 +1,147 @@
+package com.epmedu.animeal.home.presentation.ui
+
+import android.Manifest
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.platform.LocalLifecycleOwner
+import androidx.lifecycle.Lifecycle
+import com.epmedu.animeal.home.domain.PermissionStatus
+import com.epmedu.animeal.home.presentation.HomeScreenEvent
+import com.epmedu.animeal.home.presentation.viewmodel.HomeState
+import com.google.accompanist.permissions.ExperimentalPermissionsApi
+import com.google.accompanist.permissions.PermissionState
+import com.google.accompanist.permissions.isGranted
+import com.google.accompanist.permissions.rememberPermissionState
+import com.google.accompanist.permissions.shouldShowRationale
+
+@OptIn(ExperimentalPermissionsApi::class)
+@Composable
+internal fun HomePermissions(
+    homeState: HomeState,
+    onScreenEvent: (HomeScreenEvent) -> Unit,
+    content: @Composable (PermissionState) -> Unit,
+) {
+    val geolocationPermissionState = GeoLocationPermission(
+        onPermissionResponse = {
+            onScreenEvent(HomeScreenEvent.GeolocationPermissionStatusChanged(it))
+        },
+        onPermissionRequest = {
+            onScreenEvent(HomeScreenEvent.GeolocationPermissionAsked)
+        }
+    )
+    val cameraPermissionState = CameraPermission(
+        onPermissionResponse = { permissionStatus ->
+            onScreenEvent(HomeScreenEvent.CameraPermissionStatusChanged(permissionStatus))
+        },
+        onPermissionRequest = {
+            onScreenEvent(HomeScreenEvent.CameraPermissionAsked)
+        }
+    )
+    val lifecycleState = LocalLifecycleOwner.current.lifecycle.currentState
+
+    LaunchedEffect(
+        key1 = homeState.isInitialGeolocationPermissionAsked,
+        key2 = homeState.isCameraPermissionAsked,
+        key3 = lifecycleState
+    ) {
+        when {
+            !homeState.isInitialGeolocationPermissionAsked && lifecycleState.isAtLeast(Lifecycle.State.RESUMED) -> {
```

Comment:
By Lifecycle.State.Started is implied that the previous permission request dialog is still visible. If we try to call next permission request while the previous is not dismissed - it will be ignored, so we need to call it only on RESUMED state of the screen